text/0000-otlp-over-messaging-systems.md

```diff
@@ -0,0 +1,110 @@
+structured or unstructured, on that topic and the receiver wraps the log
+data in a configurable resource.
+
+## Internal details
+
+The default implementation must implement `otlp_proto`, meaning that the
+payload is Protobuf serialized from `ExportTraceServiceRequest` for traces,
+`ExportMetricsServiceRequest` for metrics, or `ExportLogsServiceRequest` for
+logs. If an implementation support other encodings an `encoding` field should
+be added to the configuration to make it switchable.
+
+| value | trace | metric | log | description |
+|---|---|---|---|---|
+| `otlp_proto` (default) | X | X | X | protobuf serialized from the `Export(Trace/Metric/Log)ServiceRequest` |
+| `otlp_json` | X | X | X | proto3 json representation of a `Export(Trace/Metric/Log)ServiceRequest` |
+| `jaeger_proto` | X | - | - | the payload is deserialized to a single Jaeger proto `Span` |
+| `jaeger_json` | X | - | - | the payload is deserialized to a single Jaeger JSON Span using `jsonpb` |
+| `zipkin_proto` | X | - | - | the payload is deserialized into a list of Zipkin proto spans |
+| `zipkin_json` | X | - | - | the payload is deserialized into a list of Zipkin V2 JSON spans |
+| `zipkin_thrift` | X | - | - | the payload is deserialized into a list of Zipkin Thrift spans |
+| `raw_string` | - | - | X | see `Log specific collector receiver` |
+| `raw_binary` | - | - | X | see `Log specific collector receiver` |
+
+Above you’ll find a non-exclusive list of possible encodings. Only `otlp_proto`
+is mandatory and the default.
+
+### Log specific collector receiver
+
+As the flow control is the hardest part of the implementations, adding a feature
+for reading raw log events avoids having an additional parallel
+implementation todo just that.
+
+The encoding field should be used to indicate that the data received is not the
+default `otlp_proto` data, but raw log data. The receiver should construct a valid
+OTLP message for each raw message received. Valid encoding are `raw_string`
+and `raw_binary`, that will control the type that the data will have when set
+in the `body` of the OTLP message.
+
+As the raw log data don’t have a resource attached to them, the receiver should add
+a generic resource and instrumentation library message around the raw messages. The
+instrumentation library should be set to the name of the receiver and the version to
+that of the collector.  Defining the exact resource should be done in the pipeline,
+using for example the `resourceprocessor`.
+
+## Prior art and alternatives
+
+The `kafkareceiver` and `kafkaexporter` already implement this OTLP as
+described in the OpenTelemetry Collector. The description in th OTEP
+makes both of them compliant without modification. Although it
+doesn't implement the raw logging support but this could be added
+without conflicting with the implementation.
+
+## Open questions
+
+This proposal doesn't take advantage of attributes that some systems
```

Comment:
+1 I think it would be good to put encoding and signal type in there.

Reply:
+1.   I think perhaps having a standard convention for knowing what signal is coming in

(either force separate channels for each signal, *OR* a standard header of some sort to disambiguate) would be good.  OTLP effectively has this in their GRPC definition via the separate services.

Reply:
Well, gRPC can handle this over the same endpoint (a separate service is just a path in the HTTP/2 header).

I'm leaning toward the CloudEvents proposal, where you can have multiple (at least have the possibility to) types over one topic.